axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveToolService.java

```diff
@@ -426,7 +426,7 @@ public String filterJournalPartnerCompatibleType(Move move) {
     if (journal != null
         && journal.getCompatiblePartnerTypeSelect() != null
         && !journal.getCompatiblePartnerTypeSelect().isEmpty()) {
-      StringBuilder compatiblePartnerDomain = new StringBuilder("self.id IN (");
+      StringBuilder compatiblePartnerDomain = new StringBuilder("AND self.id IN (");
```

Comment:
I think it's better if the `AND` is added outside of this method, so it can be used independently